packages/provider/src/wallet-request-handler.ts

```diff
@@ -177,70 +177,35 @@ export class WalletRequestHandler implements ExternalProvider, JsonRpcHandler, P
         case 'eth_signTransaction': {
           // https://eth.wiki/json-rpc/API#eth_signTransaction
           const [transaction] = request.params
-          const sender = transaction.from.toLowerCase()
+          const sender = ethers.utils.getAddress(transaction.from)
 
-          // TODO: refactor/move
           // TODO: use prompter
 
-          // TODO: just move signTransactions to the interface, and we done..
-
-          // const address = await signer.getAddress()
-
           if (this.prompter === null) {
             // ...
           }
 
-          const sig = signer.signTransaction(transaction)
-          response.result = {
-            raw: sig,
-            tx: transaction
+          if (sender === await signer.getAddress()) {
+            // const signed = await signer.signTransactions(transaction)
+            response.result = await signer.signTransactions(transaction)
```

Comment:
Is the return type of signTransactions in the format that eth_signTransaction json-rpc response type?

Reply:
or if we assume its a unique format, perhaps we can leave a comment here, to explain? would be helpful to future reader

Reply:
Yes, the idea is that this format is not meant to be handled by an RPC node directly, instead it only works if the transaction is later relayed using this Sequence handler